README.md

```diff
@@ -137,82 +137,179 @@ The trace configuration additionally exposes the `samplingRate` option which set
 
-The API is exposed by the `agent` returned by a call to `start`:
+### Accessing the API
 
-```javascript
-  var agent = require('@google/cloud-trace').start();
-```
-
-For child spans, you can either use the `startSpan` and `endSpan` API, or use the `runInSpan` function that uses a callback-style. For root spans, you must use `runInRootSpan`.
-
-### Start & end
-
-To start a new child span, use `agent.startSpan`. Each span requires a name, and you can optionally specify labels.
+Calling the `start` function returns an instance of `TraceApi`, which provides an interface for tracing:
 
 ```javascript
-  var span = agent.startSpan('name', {label: 'value'});
+  var traceApi = require('@google/cloud-trace').start();
 ```
 
-Once your work is complete, you can end a child span with `agent.endSpan`. You can again optionally associate labels with the span:
+It can also be retrieved by subsequent calls to `get` elsewhere:
 
 ```javascript
-  agent.endSpan(span, {label2: 'value'});
+  // after start() is called
+  var traceApi = require('@google/cloud-trace').get();
 ```
 
-Note that the labels parameter must be an object. Other types (e.g. string) will be silently ignored.
+The object returned by both of these calls is guaranteed to have the surface described below, even if the agent is disabled.
+
+### The `TraceApi` Object
+
+A `TraceApi` instance, in short, provides functions that facilitate the following:
+
+- Creating trace spans and add labels to them.
+- Getting information about how the trace agent was configured in the current application.
+- Parsing and serializing trace contexts to support distributed tracing between microservices.
+- Binding callbacks and event emitters in order to propagate trace contexts across asynchronous boundaries.
+
+In addition to the above, `TraceApi` also provides a number of well-known label keys and constants through its `labels` and `constants` fields respectively.
+
+#### Trace Spans
+
+These functions provide the capability to create trace spans, add labels to them, and close them. `transaction` and `childSpan` are instances of `Transaction` and `ChildSpan`, respectively.
+
+* `TraceApi#api.runInRootSpan(options, fn)`
+  * `options`: [`TraceOptions`](#trace-span-options)
+  * `fn`: `function(?Span): any`
+  * Returns `any`
+  * Attempts to create a root span, run the given callback, and pass it a `Span` object if the root span was successfuly created. Otherwise, the given function is run with `null` as an argument. This may be for one of two reasons:
+    * The trace policy, as specified by the user-given configuration, disallows a root span from being created under the current circumstances.
+    * The trace agent is disabled, either because it wasn't started at all, started in disabled mode, or started in an environment where the GCP project ID could not be obtained.
```

Comment:
I would leave the last element of this list more general. Maybe: "or encountered an initialization error"?